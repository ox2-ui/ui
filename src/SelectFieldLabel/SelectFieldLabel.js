import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin-top: 15px;
`;

const Label = styled.div`
  font-size: 15px;
  font-family: Roboto Condensed, sans-serif;
  color: #aaa;
  padding-bottom: 5px;
`;

/**
 * SelectFieldLabel Component
 */
const SelectFieldLabel = ({ children, text }) => (
  <Wrapper>
    <Label>{text}</Label>
    {children}
  </Wrapper>
);


SelectFieldLabel.propTypes = {
  children: PropTypes.node.isRequired,
  text: PropTypes.string.isRequired,
};

export default SelectFieldLabel;
