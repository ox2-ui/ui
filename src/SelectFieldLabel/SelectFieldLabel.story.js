import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SelectFieldLabel from './SelectFieldLabel';

storiesOf('SelectFieldLabel', module)
  .add('default', () => (
    <SelectFieldLabel text="My label">
      <input type="text" />
    </SelectFieldLabel>
  ));
