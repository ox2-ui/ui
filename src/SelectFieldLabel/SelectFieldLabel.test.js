/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SelectFieldLabel from './SelectFieldLabel';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <SelectFieldLabel text="My label">
      <input type="text" />
    </SelectFieldLabel>
  );

  expect(wrapper).toMatchSnapshot();
});
