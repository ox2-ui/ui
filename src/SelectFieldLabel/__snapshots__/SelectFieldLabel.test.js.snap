exports[`test renders correctly 1`] = `
<SelectFieldLabel
  text="My label">
  <styled.div>
    <div
      className="eWjddY">
      <styled.div>
        <div
          className="fpAfxb">
          My label
        </div>
      </styled.div>
      <input
        type="text" />
    </div>
  </styled.div>
</SelectFieldLabel>
`;
